woocommerce/woocommerce-ios
# Add product from image: Add analytics

Closes: #10240 


## Description

This PR adds tracking for the feature following the events listed in pe5sF9-1H4-p2.

I'm tracking the product creation source in every event, but I'm not sure if it's necessary or redundant. Please let me know if it should be removed.

Regarding the event `add_product_from_image_continue_button_tapped`, the property `has_generated_details` is tracked by checking if suggested content is applied to either the product name or description. I'm not sure if my implementation matches what you planned @jaclync - let me know if something should be updated.


## Testing instructions

- Log in to a WPCom site.
- Switch to the Products tab and tab the "+" button to create a new product.
- If the site has no product, select the manual option in the presented action sheet.
- Select the physical product in the product type action sheet.
- When the add product details from image form is displayed, notice in Xcode console: `🔵 Tracked add_product_from_image_displayed`. The source property should reflect the entry point of the product creation flow.
- Add an image to the form. After the text scan is done, if it's complete, the success event should be tracked: `🔵 Tracked add_product_from_image_scan_completed` with correct properties. Otherwise, the failure event should be tracked: `🔵 Tracked add_product_from_image_scan_failed` with the error details.
- If the text scan was successful, the product detail generation will start. If it succeeds, the success event should be tracked: `🔵 Tracked add_product_from_image_details_generated` with correct properties. Otherwise, the failure event should be tracked: `🔵 Tracked add_product_from_image_detail_generation_failed`.
- Tap the Continue button. The event `add_product_from_image_continue_button_tapped` should be tracked with correct properties.

## Screenshots

N/A

---
